crowdin.yml

```diff
@@ -0,0 +1,28 @@
+"project_id_env": "CROWDIN_PROJECT_ID"
+"api_token_env": "CROWDIN_PERSONAL_TOKEN"
+"base_path": "."
+
+"preserve_hierarchy": true
+
+"files": [
+  {
+    "source": "/src/Files.Uwp/MultilingualResources/Files.en-GB.xlf",
```

Comment:
Is this correct?

Reply:
I assumed that this is the source language of the project. There are no other files with the English language strings.

Reply:
@yaichenbaum thanks for the explanation! Can we create a corresponding `.xlf` file for the `Strings/en-US/Resources.resw` file? Then we will be able to choose `Files.en-US.xlf` as the source file and everything will work perfectly.

Later, we can add a generation of resw files to the CI, so with new translations, the Strings will also be updated.

Reply:
I don't think the current setup supports that, the xlf files are generated based on the en-us resource file.

Reply:
Hi @yaichenbaum,

just checked everything once more and it looks like the current config should work very well. 

The source language of the Crowdin project will be "English" or "English, United States" (it's up to you what to consider to choose).

Then, Crowdin Action will upload the `Files.en-GB.xlf` file as a source language file (In Crowdin UI it will be just `Files.xlf`). As a string for translation Crowdin considers only the `trans-unit` > `source` node and here this node contains the correct language - English, United States.

Also, the "English, United Kindom" language will be added as a target language. This will allow us to upload translations from the `Files.en-GB.xlf` `trans-unit` > `target` for this language (I noticed that some targets here are different from the source node).

As a result, Crowdin will create a PR with all the target languages as shown in the example PR.

Reply:
Being that en-GB is not the primary language, I think it would cause some issues if we set that as the source.

Reply:
Sounds good, what do you need from us to finish up this pull request?

Reply:
@yaichenbaum before merging the PR you need to do the following things:

- Create a new Crowdin Project:
  - As a source language select the English, United States.
  - Fill in other target languages (as a reference, you can use my [demo project](https://crowdin.com/project/files-demo), it should be easier to search languages by names).
- Generate a new Personal Access Token in your Crowdin account settings ([Crowdin docs](https://support.crowdin.com/account-settings/#api), [Crowdin Enterprise docs](https://support.crowdin.com/enterprise/personal-access-tokens/)).
- Generate a new GitHub Token (only the `repo` scope is required).
- Create the Actions secrets for this repository:
  - `CROWDIN_PERSONAL_TOKEN` - the personal access token generated in Crowdin.
  - `CROWDIN_PROJECT_ID` - ID of the Crowdin project.
  - `GH_TOKEN` - the personal access token generated in GitHub.

After that, we can merge this PR.

Reply:
Can I use the GitHub integration instead of manually creating the tokens?

Reply:
@yaichenbaum since VCS integrations in Crowdin use API to commit files but not Git, each file is committed separately. We recommend squashing commits when merging a localization branch to keep the master branch history clean and uncluttered.

Reply:
@yaichenbaum please check the "Languages Mapping" section of the [Advanced settings](https://support.crowdin.com/vcs-integrations-configuring-online/#advanced-settings). Some languages require configuring code mapping since the Crowdin **locale code** is not the same as some codes used in your app.

It's recommended to configure it via UI.

There is a list of codes that should be mapped for `%locale%`:

```
"af": "af",
"ar": "ar",
"bg": "bg",
"ca": "ca",
"da": "da",
"el": "el",
"ka": "ka",
"zh-CN": "zh-Hans",
"zh-TW": "zh-Hant",
"vi": "vi",
"ta": "ta"
```

Reply:
Squashing helps for GitHub but we still get notifications from the webhook for each commit. It would be great if Crowdin committed everything together.

Reply:
I opened a PR with the changes